src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -1469,7 +1469,7 @@ private void UpdateWindowProperties(WindowProperties newProperties, bool forceCh
                 {
                     style &= ~(fullDecorationFlags | WindowStyles.WS_THICKFRAME);
 
-                    if (newProperties.Decorations == SystemDecorations.BorderOnly && newProperties.WindowState != WindowState.Maximized && newProperties.IsResizable)
+                    if (newProperties.Decorations == SystemDecorations.BorderOnly && newProperties.IsResizable)
```

Comment:
This breaks `BorderOnly` maximized windows (when `ExtendClientAreaToDecorationsHint == false`) by adding a very thick border around them, which shouldn't be present when maximized.